waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -609,14 +609,18 @@
 (defn- get-k8s-event-deployment-error
   "Extract a deployment error if the given service includes a k8s event that is preventing
    deployment."
-  [{:keys [instances k8s/events k8s/namespace task-stats]}]
-  (let [{:keys [message reason]} (last events)
+  [{:keys [id instances k8s/events k8s/namespace task-stats]}]
+  (let [{:keys [message reason] :as latest-event} (last events)
         running-count (:running task-stats)]
     (when (and (number? instances)
                (pos? instances)
                (= 0 running-count)
                (= reason "FailedCreate"))
-      (prettify-failed-create-error message namespace))))
+      (let [details-fn (constantly {:k8s-event latest-event})
+            error-fn (fn [msg] (prettify-failed-create-error msg namespace))
+            deployment-error (create-service-deployment-error "k8s-event" message error-fn details-fn)]
```

Comment:
Does this need to be a `when-let`?

Reply:
I don't think so. All the items in this `let` block are unconditional; they're just formatting. The condition happens earlier, but that `when`'s expression doesn't return a value I would want to assign conditionally.